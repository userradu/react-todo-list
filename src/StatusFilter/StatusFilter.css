.filter-option {
  background-color: transparent;
  border: none;
  color: #428BCA;
}

.filter-option:hover {
  cursor: pointer;
}

.filter-option.selected {
  background-color: #428BCA;;
  color: #fff;
}
