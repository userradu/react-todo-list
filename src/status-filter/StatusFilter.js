import React from 'react';
import './StatusFilter.css';

function StatusFilter(props) {
  return (
    <span className={props.className}>
      <button
        className={"filter-option" + (props.currentStatus === props.status ? ' selected' : '')}
        onClick={() => props.onStatusFilterChanged(props.status)}>
        {props.status}
      </button>
    </span>
  );
}

export default StatusFilter;
