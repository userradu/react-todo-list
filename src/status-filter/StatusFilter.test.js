import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import StatusFilter from './StatusFilter';

describe('<StatusFilter />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <StatusFilter
        className="customClass"
        currentStatus="All"
        status="All"
        onStatusFilterChanged={() => { }} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls prop.onStatusFilterChanged() with prop.status as argument', () => {
    const onStatusFilterChanged = sinon.spy();
    const status = 'All';

    const wrapper = shallow(<StatusFilter
      className="customClass"
      currentStatus="All"
      status={status}
      onStatusFilterChanged={onStatusFilterChanged} />);

    wrapper.find('button').simulate('click');

    expect(onStatusFilterChanged.calledWith(status)).toBe(true);
  });

  it("it doesn't add the 'selected' class if the current status is equal to its own status", () => {
    const wrapper = shallow(<StatusFilter
      className="customClass"
      currentStatus="All"
      status="Complete"
      onStatusFilterChanged={() => { }} />);

    expect(wrapper.find('button').hasClass('selected')).toBe(false);
  });

  it("it adds the 'selected' class if the current status is equal to its own status", () => {
    const wrapper = shallow(<StatusFilter
      className="customClass"
      currentStatus="All"
      status="All"
      onStatusFilterChanged={() => { }} />);

    expect(wrapper.find('button').hasClass('selected')).toBe(true);
  });

});
