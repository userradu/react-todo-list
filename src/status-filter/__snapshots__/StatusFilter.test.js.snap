// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusFilter /> renders correctly 1`] = `
<span
  className="customClass"
>
  <button
    className="filter-option selected"
    onClick={[Function]}
  >
    All
  </button>
</span>
`;
